src/Symfony/Component/Security/Guard/Authenticator/AbstractFormLoginAuthenticator.php

```diff
@@ -60,6 +60,8 @@ public function onAuthenticationFailure(Request $request, AuthenticationExceptio
      * @param string         $providerKey
      *
      * @return RedirectResponse
+     *
+     * @deprecated since version 3.1, to be removed in 4.0
```

Comment:
the same is true here about the fact that child classes must implement it.

Reply:
Will revert